### PR TITLE
feat: adds ts-jest-mock to created type inferred mocks

### PR DESCRIPTION
**Pull request by lbenie**
## Description

Fixes (issue #2020)

I have added [ts-jest-mock](https://github.com/lbenie/ts-jest-mock), a library util helper to `react-testing` which provides a function `createMock` and a type `JestMock` to help with the DX and jest.

I'm simply re-exporting the function and type so it can be used everywhere at in Shopify.

## Type of change

- [ ]  Patch: Bug (non-breaking change which fixes an issue)
- [x] react-testing Minor: New feature (non-breaking change which adds functionality)
- [ ]  Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [ ] I have added a changelog entry, prefixed by the type of change noted above (Documentation fix and Test update does not need a changelog as we do not publish new version)

### CODE REVIEW

File: packages/react-testing/src/index.ts [hunk 1]
```diff
@@ -4,3 +4,5 @@ export type {Node} from './types';
 export * from './mount';
 export * from './destroy';
 export * from './toReactString';
+export {createMock} from 'ts-jest-mock';
```
Comment (TheMallen) [inline packages/react-testing/src/index.ts:7]: I don't like us adding jest-specific stuff to this entrypoint, currently everything jest related is cordoned off in the matchers one.

File: packages/react-testing/package.json [hunk 2]
```diff
@@ -33,7 +33,8 @@
   "dependencies": {
     "@shopify/useful-types": "^3.0.4",
     "jest-matcher-utils": "^26.6.2",
-    "react-reconciler": "^0.26.2"
+    "react-reconciler": "^0.26.2",
+    "ts-jest-mock": "^1.0.1"
```
Comment (TheMallen) [inline packages/react-testing/package.json:37]: if this is just some external lib why not just use it directly when needed?

**Review by TheMallen — CHANGES_REQUESTED** [review]
I'm not sure about this, I don't like adding jest-dependant stuff to the main entrypoint, and I don't see why folks can't just import this as needed

**Comment by lbenie** [comment]
> I'm not sure about this, I don't like adding jest-dependant stuff to the main entrypoint, and I don't see why folks can't just import this as needed

Thanks for the comment, it is a valid concern and I hear ya. 

I agree that lib could be imported directly when needed, I created this PR after a talk I had with @caropinzonsilva where we could possibly add this here instead so it would be the only entry point for that.

The only benefit I can think of is that the official React testing guides are using jest in their examples and that's the default test runner as well with CRA. With that being said, unless we plan on switching to a different test runner, it helps in the sense that this becomes the entry point to empower devs with more DX and type inferred jest mocks.

I'm still figuring out how we operate here (I'm new!).

thanks again for the feedback let me know your thoughts.

**Pull request closed without merging**